internal/git/parser.go

```
@@ -0,0 +1,74 @@
+package git
+
+import (
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+var (
+	// (user)?@host:path.git
+	scpGithubUrlRegexp = regexp.MustCompile(`^(([\w._-]+)@)?([\w.-]+):(.*)$`)
+
+	validTransportSchemas = map[string]bool{
+		"ssh":     true,
+		"git":     true,
+		"git+ssh": true,
+		"http":    true,
+		"https":   true,
+		"ftp":     true,
+		"ftps":    true,
+		"rsync":   true,
+		"file":    true,
+	}
+)
+
+func ParseUrl(urlStr string) *url.URL {
+	// match standard URL & validate schema
+	parsedUrl, err := url.Parse(urlStr)
+	if err == nil && validTransportSchemas[parsedUrl.Scheme] {
+		return parsedUrl
+	}
+
+	// match SCP URL e.g. user@host:path
+	match := scpGithubUrlRegexp.FindStringSubmatch(urlStr)
+	if match == nil {
+		// no match for standard URL or SCP URL
+		// return a local URL
+		return &url.URL{
```

Added this here to be aligned with the [functionality of git-urls](https://github.com/whilp/git-urls/blob/4a18977c6eecbf4ce0ca1e486e9ba77072ba4395/urls.go#L63), we could instead throw an error, wdyt?

---

I think it would probably be better to throw an error, I agree.

---

Testing it with an error, I realised that this actually covers the use case for local files (e.g. /path/to/repo) which is not captured by the regex and quite difficult to test on its own. I will leave it like this if you agree, it also aligns with the other unit tests in origin_url & project_name.